EpiAware/src/EpiAwareUtils/SafeNegativeBinomial.jl

```diff
@@ -0,0 +1,140 @@
+@doc raw"
+Create a Negative binomial distribution with the specified mean that avoids `InExactError`
+when the mean is too large.
+
+
+# Parameterisation:
+We are using a mean and cluster factorization of the negative binomial distribution such
+that the variance to mean relationship is:
+
+```math
+\sigma^2  = \mu + \alpha^2 \mu^2
+```
+
+The reason for this parameterisation is that at sufficiently large mean values (i.e. `μ > 1 / α`) `α` is approximately equal to the
+standard fluctuation of the distribution, e.g. if `α = 0.05` we expect typical fluctuations of samples from the negative binomial to be
+about 5% of the mean when the mean is notably larger than 20. Otherwise, we expect approximately Poisson noise. In our opinion, this
+parameterisation is useful for specifying the distribution in a way that is easier to reason on priors for `α`.
+
+# Arguments:
+
+- `μ`: The mean of the Negative binomial distribution.
+- `α`: The cluster factor of the Negative binomial distribution.
+
+# Returns:
+
+- A `SafeNegativeBinomial` distribution with the specified mean.
+
+# Examples:
+
+```jldoctest SafeNegativeBinomial
+using EpiAware, Distributions
+
+bigλ = exp(48.0) #Large value of λ
+α = 0.05
+d = SafeNegativeBinomial(bigλ, α)
+# output
+EpiAware.EpiAwareUtils.SafeNegativeBinomial{Float64}(μ=7.016735912097631e20, α=0.05)
+```
+
+```jldoctest SafeNegativeBinomial
+cdf(d, 2)
+# output
+0.0
+```
+
+```jldoctest SafeNegativeBinomial
+logpdf(d, 100)
+# output
+-16556.546939786767
+```
+
+```jldoctest SafeNegativeBinomial
+mean(d)
+# output
+7.016735912097631e20
+```
+
+```jldoctest SafeNegativeBinomial
+var(d)
+# output
+1.2308645715030148e39
+```
+"
+struct SafeNegativeBinomial{T <: Real} <: DiscreteUnivariateDistribution
+    μ::T
+    α::T
+
+    function SafeNegativeBinomial{T}(μ::T, α::T) where {T <: Real}
+        return new{T}(μ, α)
+    end
+end
+
+#Outer constructors make AD work
+function SafeNegativeBinomial(μ::T, α::T) where {T <: Real}
+    return SafeNegativeBinomial{T}(μ, α)
+end
+
+SafeNegativeBinomial(μ::Real, α::Real) = SafeNegativeBinomial(promote(μ, α)...)
+
+# helper function
+function _negbin(d::SafeNegativeBinomial)
+    μ² = d.μ^2
+    ex_σ² = d.α^2 * μ²
```

Comment:
I think we should directly paramterise to `NegativeBinomial` and pass to this from `NegativeMeanClust`.

Reply:
Yeah this makes sense.